pennylane/interfaces/autograd.py

```diff
@@ -67,14 +68,39 @@ def gradient_product(g):
                     nested Sequence[float]: vector-Jacobian product, arranged
                     into the nested structure of the input arguments in ``args``
                 """
+                diff_args = []
+                diff_indices = None
+                non_diff_indices = set()
+
+                for arg, variable_inds in zip(args, self.arg_vars):
```

Comment:
is `arg_vars` a poor name choice? It seems that that object actually contains variable indices :thinking:

Reply:
I agree, I was reluctant to change it here as it felt out of scope (it's used in `BaseQNode`, `JacobianQNode`, and `PassthruQNode` I believe)

Reply:
Wait, actually, I was wrong. `self.arg_vars` contains a nested list of `Variable` objects, in the same shape as the input arguments. So if anything, my use of `variable_inds` is incorrect here. I'll rename it to `variables`.

Reply:
Have renamed the loop variable to `arg_variable`, rather than `variable_inds` (which implies an iterable of integers).